### ISSUE TITLE
Issue with trailing / redirect

### ISSUE THREAD

**Issue by faizan-glitch** [uptrace/bunrouter#89]
BunRouter is redirecting a POST request without a trailing to the GET Handler

I have the following code 
![image](https://user-images.githubusercontent.com/58991337/233042561-754f831f-b0ae-4b22-bceb-77a80e4f36ad.png)

If i send a POST request to **/accounts/** then it works perfectly fine
But if send a POST request to **/accounts** then it redirects to GET /accounts
I have attached some screenshots below
As it can be seen that it works fine if i have a trailing slash
![image](https://user-images.githubusercontent.com/58991337/233041246-93ca9182-02f9-4376-86a9-4789961dce88.png)

but if I remove the trailing slash I get this
![image](https://user-images.githubusercontent.com/58991337/233041534-89b7d806-6cbc-43f4-888f-f17ba1c61144.png)

Notice how I get method not allowed that is because I am getting redirected to a GET handler for /accounts/ which doesn't exists
![image](https://user-images.githubusercontent.com/58991337/233041862-868b129a-63b7-4612-bbf6-99ca6c756858.png)

POST Request to /accounts should get redirected to POST handler of /accounts/ instead of GET handler of /

**Comment by aikchun** [comment]
@faizan-glitch For this case, you can just omit the slash.
```
router.WithGroup("/accounts", func(g *bunrouter.CompatGroup) {
    // g.POST("/", accountHandler.Signup)
    g.POST("", accountHandler.Signup)
}
```
Like this.
Of course, this present the issue with that if you were to request `POST -  /accounts/` you will still get `GET /accounts/`

**Comment by faizan-glitch** [comment]
@aikchun yes, i realize that i can omit the trailing slash, but that is not the expected behaviour, because I cannot make sure that the frontend devs will always omit it. 

As of now I have added two seperate routes, one with the trailing slash another without the trailing slash to get around this. but it is very weird.

is there any particular reason why Bunrouter  by default redirects towards GET instead of whatever is the correct HTTP method?

**Comment by aikchun** [comment]
> is there any particular reason why Bunrouter by default redirects towards GET instead of whatever is the correct HTTP method?

I am not sure myself. I was looking to contribute to the project through your issue so I am kinda new to the project and contributing to OSS in general. It does seem like it is handling trailing slashes [in a way](https://github.com/uptrace/bunrouter/blob/1a7449219a1d96e6808a243646266b1675380a4e/router.go#L96), I'm just not sure how.

**Comment by malekim** [comment]
What's going on with that issue? Is the project still maintained?
